src/rgw/rgw_rados.cc

```diff
@@ -5195,17 +5195,13 @@ int RGWRados::Bucket::List::list_objects(int max, vector<rgw_bucket_dir_entry> *
   result->clear();
 
   rgw_obj_key marker_obj(params.marker.name, params.marker.instance, params.ns);
-
-  rgw_obj_key end_marker_obj;
-  rgw_obj_index_key cur_end_marker;
-  if (!params.ns.empty()) {
-    end_marker_obj = rgw_obj_key(params.end_marker.name, params.end_marker.instance, params.ns);
-    end_marker_obj.ns = params.ns;
-    end_marker_obj.get_index_key(&cur_end_marker);
-  }
   rgw_obj_index_key cur_marker;
   marker_obj.get_index_key(&cur_marker);
 
+  rgw_obj_key end_marker_obj(params.end_marker.name, params.end_marker.instance,
+                             params.ns);
+  rgw_obj_index_key cur_end_marker;
+  end_marker_obj.get_index_key(&cur_end_marker);
```

Comment:
Hopefully this patch won't interfere with the fix for [the issue 18331](http://tracker.ceph.com/issues/18331). The orphan tool [doesn't use](https://github.com/rzarzynski/ceph/blob/022575c73784d4934672079f4eb3ad6a9307c311/src/rgw/rgw_orphan.cc#L488-L490) the `end_marker` parameter, so `cur_end_marker_valid` should still be `false`, and thus inhibit any changes related to handling of `cur_end_marker` in the loop.